rpcs3/Emu/RSX/VK/VKGSRender.cpp

```diff
@@ -1698,6 +1712,75 @@ void VKGSRender::do_local_task()
 		if (!cb->pending)
 			m_last_flushable_cb = -1;
 	}
+
+#ifdef _WIN32
+
+	switch (m_frame->get_wm_event())
+	{
+	case wm_event::none:
+		break;
+	case wm_event::geometry_change_notice:
+	{
+		//Stall until finish notification is received. Also, raise surface dirty flag
+		u32 timeout = 1000;
+		bool handled = false;
+
+		while (timeout)
+		{
+			switch (m_frame->get_wm_event())
+			{
+			default:
+				break;
+			case wm_event::window_resized:
+				handled = true;
+				present_surface_dirty_flag = true;
+				break;
+			case wm_event::window_moved:
+				handled = true;
+				break;
+			case wm_event::geometry_change_in_progress:
+				timeout += 10; //extend timeout to wait for user to finish resizing
+				break;
+			}
+
+			if (handled)
+				break;
+			else
+			{
+				//wait for window manager event
+				std::this_thread::sleep_for(10ms);
+				timeout -= 10;
+			}
+		}
+
+		if (!timeout)
+		{
+			LOG_ERROR(RSX, "wm event handler timed out");
+		}
+
+		break;
+	}
+	case wm_event::window_resized:
+	{
+		LOG_ERROR(RSX, "wm_event::window_resized received without corresponding wm_event::geometry_change_notice!");
+		std::this_thread::sleep_for(100ms);
+		break;
+	}
+	}
+
+#else
+
+	const u32 frame_width = m_frame->width();
```

Comment:
shouldn't it be `client_width` or something?

Btw, [Qt 5.10 naively supports Vulkan](http://blog.qt.io/blog/2017/06/06/vulkan-support-qt-5-10-part-1/), though not sure how applicable/helpful it is.

Reply:
1. Yes, should be client_width for consistency.

2. As for Qt 5.10, it depends on whether they choose to handle the swapchain themselves or not. I doubt they'll choose to do so. As long as rpcs3 is managing its own swapchain and swapchain images, it will always be necessary to notify the renderer that the window manager is about to break things - something Qt has refused to do in the past so we'll have to wait and see.

Reply:
We're in a little bit of a predicament with Qt libs on Windows, most people just replace the rpcs3.exe, so when we move to Qt 5.10, most people would just replace rpcs3.exe, leaving the old Qt dlls. I wonder if Qt would throw an error if using old libs.

But yeah, we definitely plan to update to 5.10, mostly because of the Ssl issues Megamouse faced.

Reply:
@hcorion ssl issues were not related to Qt version. It's just the issue with cryptography libs. It's illegal for Qt in some countries to bundle openSSL, so it doesn't. Instead the lib is expected to be either preinstalled on user's OS or provided by the dev. IMHO, totally not worth the hassle for that single feature and I think that Dolphin and PCSX2 approach of keeping compat/hacks DB in-source is better (since the db will be synced with the build that way).

@kd-11 I think [the third part](http://blog.qt.io/blog/2017/07/03/vulkan-support-qt-5-10-part-3/) talks about managing swapchain, didn't fully read it though.

Reply:
I just checked the class documentation. Its very bare bones but in theory should make this easier to do since they only deal with the window management. I'll have to check the source for implementation details as well. It will be a while before 5.11 is ready and a while before the emu migrates to it. Some of the design principles clash directly with optimization done for rpcs3 but it may be possible with some work to work around those constraints.